### PR TITLE
Revert the changes added to reuse fail nodes per on-fail clause

### PR DESCRIPTION
**Pull request by gabilang**
reported issue: https://github.com/ballerina-platform/ballerina-lang/issues/39142

Fixes #<Issue Number>

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

**Review by Nadeeshan96 — APPROVED** [review]
LGTM

File: tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/expressions/checkedexpr/CheckedExprErrorReturnTypeTest.java [hunk 1]
```diff
@@ -1,59 +0,0 @@
-/*
```
Comment (warunalakshitha) [inline tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/expressions/checkedexpr/CheckedExprErrorReturnTypeTest.java:1]: Why we remove this tests?

Comment (gabilang) [reply tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/expressions/checkedexpr/CheckedExprErrorReturnTypeTest.java:1]: These tests has been added to validate the behaviour after reusing the temporary error variables which are generated while desugaring the checked expression. Anyway these are valid tests regardless of the changes. I thought of adding along with the fix going to be added to the issue https://github.com/ballerina-platform/ballerina-lang/issues/39369. We can leave it without removing as well. WDYT?

Comment (warunalakshitha) [reply tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/expressions/checkedexpr/CheckedExprErrorReturnTypeTest.java:1]: +1. Lets remove it and add with proper fix.

**Review by warunalakshitha — APPROVED** [review]

**Pull request merged**